mesonbuild/build.py

```diff
@@ -2339,10 +2344,25 @@ def process_kwargs(self, kwargs, backend):
                 # the list index of that item will not be installed
                 self.install_dir = typeslistify(kwargs['install_dir'], (str, bool))
                 self.install_mode = kwargs.get('install_mode', None)
+                # If install_tag is a string we set the same tag on all outputs.
+                # Otherwise it must be a list of strings matching the list of outputs.
+                # If the list of tags is shorter, it is assumed that remaining
+                # outputs have no tag.
+                self.install_tag = kwargs.get('install_tag', [])
+                if isinstance(self.install_tag, str):
+                    self.install_tag = [self.install_tag] * len(self.outputs)
+                else:
+                    self.install_tag = stringlistify(self.install_tag)
+                    missing = len(self.outputs) - len(self.install_tag)
+                    if missing < 0:
+                        raise InvalidArguments('install_tag list cannot have more items than outputs')
+                    if missing > 0:
+                        self.install_tag += [None] * missing
```

Comment:
Please no, this is going to make used `typed_kwargs` for this miserable, since `foo` and `[foo]` will have different meanings, which Meson generally does not do. Can we instead do something like:
```python
install_tag = extract_as_list(kwargs, 'install_tag')
if len(install_tag) == 1:
   self.install_tag = [install_tag] * len(install_tag)
elif len(install_tag) != len(self.outputs):
   # raise an error message
else:
  self.install_tag = install_tag
```

If you want something to be untagged you could pass `false` like we do for `install_dir`, so at least those are symaterical, but we're also not providing different behavior for `['devel']` and `'devel'`. I think this is important because Meson generally does not distinguish between an scalar and an array of length 1.

Reply:
You're right, better stick to the same behaviour as `install_dir` which was my initial intention. I changed that code, what do you think?